modules/mapbox-layers/src/deck-layer.js

```diff
@@ -23,29 +33,64 @@ export default class DeckLayer {
     };
   }
 
-  onAdd(map, gl) {
-    // console.log('onAdd', map, gl);
+  _getDeckInstance(gl) {
+    // Only create one deck instance per context
+    let deck = deckInstances.get(gl);
+
+    if (!deck) {
+      deck = new Deck({
+        // TODO - this should not be needed
+        canvas: 'deck-canvas',
+        width: '100%',
+        height: '100%',
+        controller: false,
+        _customRender: true,
+        _mapboxLayers: new Set(),
```

Comment:
Hmm, I am not sure about this. It works with the current "sloppy props". But I would prefer us not having a `Deck` prop named `mapbox...`? Wonder if can leverage/improve the `userData` mechanism?